siliconcompiler/schema/schema_cfg.py

```diff
@@ -1971,7 +1977,7 @@ def schema_option(cfg):
 
     # Compilation
     scparam(cfg, ['option', 'mode'],
-            sctype='str',
+            sctype='enum',
```

Comment:
Since these mode strings get called out in the code, I agree that it should be an enum. Otherwise, the user ends up getting weird downstream errors.